lib/surface/compiler.ex

```diff
@@ -284,20 +284,23 @@ defmodule Surface.Compiler do
   defp convert_node_to_ast(:slot, {_, attributes, children, node_meta}, compile_meta) do
     meta = Helpers.to_meta(node_meta, compile_meta)
 
-    defined_slot_names =
+    defined_slots =
       meta.caller.module
       |> Surface.API.get_slots()
-      |> Enum.map(& &1.name)
 
     # TODO: Validate attributes with custom messages
```

Comment:
@msaraiva What did you have in mind with this TODO?

Reply:
Indeed it results in an internal error as the compiler tries to convert `{:attribute_expr, " :slot_name ", %{line: 2}}` to an atom.

```console
== Compilation error in file test/xxx.exs ==
** (ArgumentError) argument error
    :erlang.binary_to_atom({:attribute_expr, " :body ", %{line: 2}}, :utf8)
    (surface 0.3.1) lib/surface/compiler.ex:485: anonymous fn/2 in Surface.Compiler.attribute_value/3
    (elixir 1.11.0) lib/enum.ex:3411: Enum.find_value_list/3
    (surface 0.3.1) lib/surface/compiler.ex:292: Surface.Compiler.convert_node_to_ast/3
    (surface 0.3.1) lib/surface/compiler.ex:212: anonymous fn/3 in Surface.Compiler.to_ast/2
    (elixir 1.11.0) lib/enum.ex:2181: Enum."-reduce/3-lists^foldl/2-0-"/3
    (surface 0.3.1) lib/surface/compiler.ex:211: Surface.Compiler.to_ast/2
    (surface 0.3.1) lib/surface/compiler.ex:331: Surface.Compiler.convert_node_to_ast/3
    (surface 0.3.1) lib/surface/compiler.ex:212: anonymous fn/3 in Surface.Compiler.to_ast/2
    (elixir 1.11.0) lib/enum.ex:2181: Enum."-reduce/3-lists^foldl/2-0-"/3
    (surface 0.3.1) lib/surface/compiler.ex:211: Surface.Compiler.to_ast/2
    (surface 0.3.1) lib/surface/compiler.ex:115: Surface.Compiler.compile/5
    (surface 0.3.1) expanding macro: Surface.sigil_H/2
    test/slot_test.exs:72: Surface.SlotTest.OuterWithNamedSlotAndProps.render/1
    (elixir 1.11.0) lib/kernel/parallel_compiler.ex:416: Kernel.ParallelCompiler.require_file/2
```